dkc/core/exceptions.py

```diff
@@ -0,0 +1,8 @@
+from django.core.exceptions import ValidationError
+
+
+class QuotaLimitedError(ValidationError):
+    """Raised when a quota would be exceeded by an increment operation."""
+
+    def __init__(self, message='Root folder size quota would be exceeded.', *args, **kwargs):
```

Comment:
Will this message be directly displayed to any users? If not, maybe say:
> Tree size quota would be exceeded

instead, since that's more accurate and helpful to developers.